### ISSUE TITLE
Support Modules

### ISSUE THREAD

**Issue by posva** [posva/vuexfire#7]
Currently, vuexfire only let you declare refs in the state.
In bigger applications, you may need to to use modules.

**Comment by zoomclub** [comment]
Had a problem running https://github.com/niallobrien/Vue-Vuex-Firebase, which lead me to here. Suppose it is due to the following issue https://github.com/posva/vuexfire/issues/6

Looking forward then to module support for vuexfire :)

**Comment by posva** [comment]
Thanks for showing some interest 😄 
It's high on my todo list.

**Comment by weotch** [comment]
Do you have a rough ETA on this?  Would you be open to a PR to add this support?  I'm thinking of an approach where the `mutations` property becomes a function that you call with with a prefix that is, generally, the module name.  LIke:

```js
var store = new Vuex.Store({
  state: {
    items: null
  },
  mutations: VuexFire.mutations('clients'),
  getters: {
    items: function (state) { return state.items }
  }
})
```

This creates mutator keys like:

```js
'VUEXFIRE/clients/objectValue'
'VUEXFIRE/clients/arrayChange'
// etc
```

Then, you use that same key when you register the bindings:

```js
new Vue({
  el: '#app',
  store: store,
  computed: Vuex.mapGetters([
    'items'
  ]),
  firebase: {
    'clients.items': db.ref('items')
  }
})
```

When the binding keys get parsed, if there is a `.` in the key, it gets `split` into module and key.

The same syntax would be used throughout, for instance: `vm.$unbind('clients.items')`.

**Comment by posva** [comment]
@weotch Thanks for your interest. I'll like to split the tests files first to make it easier to test. But we would write tests for modules in a new file anyway.
I haven't taken the time to think about the features I'd like to show.
A pessimistic ETA would be November 7

Let me think about the problem too first. After that I'll be open to PR 😄

**Comment by posva** [comment]
@weotch 
IMO the user should be able to specify the module in the key as you said:

``` js
firebase: {
  'clients.items': db.ref('items')
}
```

And vuexfire should handle the rest by splitting the key and identifying the module.
What I'm wondering is if I should add actions for each submodule or just add each action once. I need to play around with it a bit before solving that

**Comment by ryanmclaughlin** [comment]
@posva 
Any updates on this?

@weotch Did you ever get modules to work?

**Comment by posva** [comment]
@ryanmclaughlin I'll start spending time on this. I should be able to get something released by the end of the next week

**Comment by posva** [comment]
I released the version 0.3.0 with modules. Please, try it and give feedback 😄

**Comment by ryanmclaughlin** [comment]
![hugs](https://media0.giphy.com/media/u0bQN6bxXweHe/200.gif#0)
